code/modules/mob/living/simple_animal/abnormality/waw/greed_king.dm

```diff
@@ -0,0 +1,167 @@
+	if(charging)
+		return
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/Move()
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/Life()
+	. = ..()
+	if(charge_cooldown < world.time && !charging && !(status_flags & GODMODE))
+		charge_check_cooldown = world.time + charge_check_time
+		if(prob(30))
+			teleport()
+		else
+			charge_check()
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/breach_effect(mob/living/carbon/human/user)
+	..()
+	icon = 'ModularTegustation/Teguicons/64x48.dmi'
+	//Center it on a hallway
+	pixel_y = -8
+	base_pixel_y = -8
+
+	teleport()	//Let's Spaghettioodle out of here
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/teleport()
+	charge_cooldown = world.time + charge_check_time
+	if(charging)
+		return
+	var/turf/T = pick(GLOB.xeno_spawn)
+	animate(src, alpha = 0, time = 5)
+	SLEEP_CHECK_DEATH(5)
+	animate(src, alpha = 255, time = 5)
+	forceMove(T)
+	//Clear lists
+	been_hit = list()
+	SLEEP_CHECK_DEATH(2)
+	charge_check()
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/charge_check()
+	//targeting
+	var/list/possible_targets = list()
+	for(var/mob/living/carbon/human/H in view(20, src))
+		possible_targets += H
+	var/mob/living/carbon/human/target = pick(possible_targets)
+	if(!LAZYLEN(possible_targets))
+		return
+	//Start charge
+	if(ishuman(target))
+		var/dir_to_target = get_cardinal_dir(get_turf(src), get_turf(target))
+		if(dir_to_target in list(NORTH, SOUTH, WEST, EAST))
+			addtimer(CALLBACK(src, .proc/charge, dir_to_target, 0, target), 2 SECONDS)
+			return
+
+/mob/living/simple_animal/hostile/abnormality/greed_king/proc/charge(move_dir, times_ran, target)
+	setDir(move_dir)
+	var/stop_charge = FALSE
+	if(times_ran >= dash_num)
+		stop_charge = TRUE
+	var/turf/T = get_step(get_turf(src), move_dir)
+	if(!T)
+		charging = FALSE
+		return
+	if(T.density)
+		stop_charge = TRUE
+	for(var/obj/structure/window/W in T.contents)
+		W.obj_destruction()
+	for(var/obj/machinery/door/D in T.contents)
+		if(D.density)
+			stop_charge = TRUE
+	for(var/mob/living/simple_animal/hostile/abnormality/D in T.contents)	//This caused issues earlier
+		if(D.density)
+			stop_charge = TRUE
+	if(stop_charge)
+		SLEEP_CHECK_DEATH(7 SECONDS)
+		charging = FALSE
+		teleport()
+		return
+	forceMove(T)
+	//Hiteffect stuff
+	for(var/mob/living/L in range(1, T))
+		charging = TRUE
+		if(L in been_hit)
+			continue
+		been_hit+=L
+		visible_message("<span class='boldwarning'>[src] crunches [L]!</span>")
+		to_chat(L, "<span class='userdanger'>[src] rends you with it's teeth!</span>")
+		playsound(L, attack_sound, 75, 1)
+		var/turf/LT = get_turf(L)
+		new /obj/effect/temp_visual/kinetic_blast(LT)
+		if(ishuman(L))
+			var/mob/living/carbon/human/H = L
+			H.apply_damage(800, RED_DAMAGE, null, L.run_armor_check(null, RED_DAMAGE), spread_damage = TRUE)
+		else
+			L.adjustRedLoss(80)
+			if(L.stat != DEAD)	//If they're not dead, stop charge.
+				stop_charge = TRUE
```

Comment:
This won't ever work because the check for stop_charge is... well a little higher up.